src/Doctor/Doctor.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Dock\Doctor;
+
+class Doctor
+{
+    /**
+     * @var array
+     */
+    private $tasks;
+
+    /**
+     * @param array $tasks
+     */
+    public function __construct(array $tasks)
+    {
+        $this->tasks = $tasks;
+    }
+
+    public function examine($dryRun)
+    {
+        foreach ($this->tasks as $task) {
+            $task->run($dryRun);
```

Comment:
So you do not use the result of `Task::run` which is supposed to be a boolean ? Maybe returning true if all of these tests are true and then display a message and return the according result (0 or 1) from the doctor command ?